code-challenges.test.js

```diff
@@ -14,46 +14,224 @@
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        const output1 = ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+        const ouput2 = ["Temperature", "Database", "Chopsticks", "Mango"]
+
+        // Expect statements using provided inputs/outputs
+        expect(capitalizeAll(randomNouns1)).toEqual(output1)
+        expect(capitalizeAll(randomNouns2)).toEqual(ouput2)
+    })
+})
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
+// A function that accepts an array of strings
+    // Returns an array with all the words capitalized
+// Errors received:
+// SyntaxError: Missing semicolon.
+// // Final output:
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+const capitalizeAll = (array1) => {
+    // Iterate through the array
+        // Use map to return a new array
+            // With each entry being the first letter capitalized and the remainder spliced
+    return array1.map((value) => (value[0].toUpperCase()+value.substring(1)))
+}
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
-
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+// Error received: firstVowel is not defined
+describe("firstVowel" , () => {
+
+    it ("takes in a string and logs the index of the first vowel", () => {
+
+        // Use supplied test criteria
+        const vowelTester1 = "learn"
+        // Expected output: 1
+        const output1 = 1
+        const vowelTester2 = "academy"
+        // Expected output: 0
+        const output2 = 0
+        const vowelTester3 = "challenges"
+        // Expected output: 2
+        const output3 = 2
+        
+        expect(firstVowel(vowelTester1)).toEqual(output1)
+        expect(firstVowel(vowelTester2)).toEqual(output2)
+        expect(firstVowel(vowelTester3)).toEqual(output3)
+    })
+})
 
 // b) Create the function that makes the test pass.
+// A function that accepts a string 
+    //and returns the index of the first vowel
+const firstVowel = (str1) => {
+
+    // Iterate through the string
+    for (let i = 0 ; i < str1.length ; i++){
+
+        // Check for vowels at current index
+        // Future work is to add the rest of the vowels
```

Comment:
You would definitely want to consider all other vowels in this, but overall the logic looks great!